addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java

```diff
@@ -176,7 +176,7 @@ public void onServiceSubscribed(String service, boolean succeeded) {
     public void onValueReceived(String variable, String value, String service) {
 
         String oldValue = stateMap.get(variable);
-        if (value.equals(oldValue)) {
+        if (oldValue.equals(value)) {
```

Comment:
@martinvw Better but now the case `value==null && oldValue==null` will be ignored. Looking at the switch case below `value==null` is anticipated and Map.get() can return null. According to the original logic this case should be ignored.